pil-analyzer/src/pil_analyzer.rs

```diff
@@ -287,8 +288,16 @@ impl PILAnalyzer {
                     if let Some(selector) = &mut part.selector {
                         expressions.push((selector, Type::Expr.into()))
                     }
-                    for e in &mut part.expressions {
-                        expressions.push((e, Type::Expr.into()))
+                    if let Expression::ArrayLiteral(_, ArrayLiteral { items }) =
```

Comment:
It's fine to use `x $ f() in y $ g()` now, as long as `f()` and `g()` have the type `Expr[]`. So you shouldn't iterate through the items here but just request an array as type for `part.expressions`.